solr/core/src/java/org/apache/solr/schema/DenseVectorField.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * <br>
+ * Only {@code Indexed} and {@code Stored} attributes are supported.
+ */
+public class DenseVectorField extends FieldType {
+
+    static final String KNN_VECTOR_DIMENSION = "vectorDimension";
+    static final String KNN_SIMILARITY_FUNCTION = "similarityFunction";
+    
+    int dimension;
+    VectorSimilarityFunction similarityFunction;
+    VectorSimilarityFunction DEFAULT_SIMILARITY = VectorSimilarityFunction.EUCLIDEAN;
+
+    @Override
+    public void init(IndexSchema schema, Map<String, String> args){
+        this.dimension = ofNullable(args.get(KNN_VECTOR_DIMENSION))
+                .map(value -> parseDimensionParameter(value))
+                .orElseThrow(() -> new SolrException(SolrException.ErrorCode.SERVER_ERROR, "the vector dimension is a mandatory parameter"));
+        args.remove(KNN_VECTOR_DIMENSION);
+
+        this.similarityFunction = ofNullable(args.get(KNN_SIMILARITY_FUNCTION))
+                .map(value -> VectorSimilarityFunction.valueOf(value.toUpperCase(Locale.ROOT)))
+                .orElse(DEFAULT_SIMILARITY);
+        args.remove(KNN_SIMILARITY_FUNCTION);
+
+        this.properties &= ~MULTIVALUED;
+        this.properties &= ~UNINVERTIBLE;
+        
+        super.init(schema, args);
+    }
+
+    private int parseDimensionParameter(String value) {
+        try {
+            return Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "the vector dimension must be an integer");
+        }
+    }
+
+    @Override
+    public void checkSchemaField(final SchemaField field) throws SolrException {
+        super.checkSchemaField(field);
+        if (field.multiValued()) {
+            throw new SolrException(SolrException.ErrorCode.SERVER_ERROR,
+                    getClass().getSimpleName() + " fields can not be multiValued: " + field.getName());
+        }
+    }
+    
+    public List<IndexableField> createFields(SchemaField field, Object value) {
+        List<IndexableField> fields = new ArrayList<>();
+        if (field.indexed()){
+            fields.add(createField(field, value));
+        }
+        if (field.stored()){
+            fields.add(createStoredField(field, value));
+        }
+        return fields;
+    }
+
+    private IndexableField createStoredField(SchemaField field, Object value) {
+        return new StoredField(field.getName(), value.toString());
+    }
+
+    @Override
+    public UninvertingReader.Type getUninversionType(SchemaField sf) {
+        return null;
+    }
+
+    @Override
+    public ValueSource getValueSource(SchemaField field, QParser parser) {
+        throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+                "Function queries are not supported for Dense Vector fields.");
+    }
+
+    @Override
+    public void write(TextResponseWriter writer, String name, IndexableField f) throws IOException {
+        writer.writeVal(name, f.stringValue());
```

Comment:
Not sure what stringValue will return here - will it be the `Arrays.toString()` of the vector? Something else?

Reply:
This should have been resolved with latest commit